lib/active_record_shards/default_slave_patches.rb

```diff
@@ -26,14 +26,8 @@ def #{method}_with_default_slave#{punctuation}(*args, &block)
     end
 
     def columns_with_default_slave(*args, &block)
-      read_columns_from =
-        if on_slave_by_default? && !Thread.current[:_active_record_shards_slave_off]
```

Comment:
I thought that `on_slave_by_default?` would do what the PR title says: “read columns from slaves by default”.